ext/opentelemetry-ext-opentracing-shim/src/opentelemetry/ext/opentracing_shim/__init__.py

```diff
@@ -0,0 +1,289 @@
+        )
+        return ScopeWrapper(self, span_cm=span_cm)
+
+    @property
+    def active(self):
+        span = self._tracer.unwrap().get_current_span()
+        if span is None:
+            return None
+
+        wrapped_span = SpanWrapper(self._tracer, span.get_context(), span)
+        return ScopeWrapper(self, span=wrapped_span)
+        # TODO: Return a saved instance of SpanWrapper instead of constructing
+        # a new object (and the same for ScopeWrapper?).
+        # https://github.com/open-telemetry/opentelemetry-python/issues/161#issuecomment-534136274
+
+    @property
+    def tracer(self):
+        return self._tracer
+
+
+class TracerWrapper(opentracing.Tracer):
+    def __init__(self, tracer, scope_manager=None):
+        scope_manager = ScopeManagerWrapper(self)
+        super().__init__(scope_manager=scope_manager)
+        self._otel_tracer = tracer
+
+    def unwrap(self):
+        """Returns the wrapped OpenTelemetry `Tracer` object."""
+
+        return self._otel_tracer
+
+    def start_active_span(
+        self,
+        operation_name,
+        child_of=None,
+        references=None,
+        tags=None,
+        start_time=None,
+        ignore_active_span=False,
+        finish_on_close=True,
+    ):
+        span = self.start_span(
+            operation_name=operation_name,
+            child_of=child_of,
+            references=references,
+            tags=tags,
+            start_time=start_time,
+            ignore_active_span=ignore_active_span,
+        )
+        return self._scope_manager.activate(span, finish_on_close)
+
+    def start_span(
+        self,
+        operation_name=None,
+        child_of=None,
+        references=None,
+        tags=None,
+        start_time=None,
+        ignore_active_span=False,
+    ):
+        if child_of is None:
+            parent = None
+        else:
+            if isinstance(child_of, (SpanWrapper, SpanContextWrapper)):
+                # The parent specified in `child_of` is valid and is either a
+                # `SpanWrapper` or a `SpanContextWrapper`. Unwrap the `Span` or
+                # `SpanContext` to extract the OpenTracing object and use this
+                # object as the parent of the created span.
+                parent = child_of.unwrap()
+            else:
+                logger.warning(
+                    "Unknown class %s passed in child_of argument to start_span() method.",
+                    type(child_of),
+                )
+                parent = None
+                # TODO: Refuse to create a span and return `None` instead of
+                # proceeding with a `None` parent? This would cause the created
+                # span to become a child of the active span, if any, or create
+                # a new trace and make the span the root span of that trace.
+
+        # Use active span as parent when no explicit parent is specified.
+        if (
+            self.active_span is not None
```

Comment:
Unless, I'm wrong, you will end up calling `self.active_span()` even if there's already an explicit, validated `parent` (the check will be correct, but still).

Reply:
Good catch, thanks. I've changed the order of the statements so now `self.active_span()` isn't called when we have a valid parent.